nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -1038,19 +1038,16 @@ ngx_http_modsecurity_handler(ngx_http_request_t *r)
         return NGX_DECLINED;
     }
 
-    if (r->method == NGX_HTTP_POST 
-            && modsecIsRequestBodyAccessEnabled(ctx->req) ) {
 
-        /* read POST request body, should we process PUT? */
+    if (modsecIsRequestBodyAccessEnabled(ctx->req) && (r->headers_in.content_length || r->headers_in.chunked)) {
```

Comment:
just thinking, if in the non-chunked case, if the content_length doesn't match with the real body size, nginx will send back BAD REQUEST, right, which won't let the request go through modsec, just to confirm

Reply:
If the content is shorter than the content-length header said, then it will just wait until it gets the content it was promised. If the content was longer (this is what I meant by trailing data in my comment above), then it will ignore the trailing data. This behavior seems correct to me.